docs/index.rst

```diff
@@ -1,8 +1,76 @@
 ``pluggy``
 ==========
 
-The ``pytest`` plugin system
-****************************
+The five minute elevator pitch
+******************************
+
+``pluggy`` gives `users` the possibility to extend or modify the behaviour of
+a program like e.g. `pytest`_, `tox`_ or `devpi`_ by installing a `plugin` for
+that program. The plugin code will run as part of normal program execution
+changing or enhancing certain aspects of it.
+
+There are three roles in which people might interact with ``pluggy``.
+
+Publisher
++++++++++
+
+The `publisher` wants to enable an arbitrary number of `subscribers` to extend
+or modify some aspects of their programs execution. They write hook specifications,
+register them with a :py:class:`pluggy.PluginManager` and instantiate a marker
+object for the `subscriber` - a :py:class:`~pluggy.HookimplMarker` - used
+to `decorate <https://www.python.org/dev/peps/pep-0318/#current-syntax>` hook
+implementation functions. This marker is usually called `hookimpl` and
+importable from the root package of the `publisher` project. They then add
+calls to their specified hooks in appropriate phases of program execution.
+``pluggy`` will then do all the magic.
+
+**FIXME not sure what is the right advice here. Do I need to vendor in pluggy as a publisher to avoid versioning conflicts? Should I just depend on pluggy without version and hope all is good?**
+To avoid compatibility issues they should always pin their ``pluggy`` version
+to the minor version used at implementation time (at the time of writing
+(spring 2018) ``pluggy`` is not at 1.0 yet, so breaking changes should be
+expected with any minor release).
+
+They need to learn about ``pluggy`` and its capabilities and should at least
+loosely follow the development of the ``pluggy`` project.
+
+Subscriber
```

Comment:
Maybe `Provider` or `Implementer` as in plugin provider?

Reply:
implementer feels also right to me.